[Automation] native-image run fails for io.mockk:mockk-agent-jvm:1.14.9

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.14.9" ./gradlew clean test -Pcoordinates="io.mockk:mockk-agent-jvm:1.14.9"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26073520164/job/76659718634
Last 300 lines of the log:
```console
--add-exports=java.base/sun.reflect.generics.repository=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.scope=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.tree=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.jca=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.rsa=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.ssl=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.x509=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.text.spi=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.calendar=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.cldr=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.locale.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.locale=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.resources=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.spi=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util=org.graalvm.nativeimage.builder \
--add-exports=java.management/com.sun.jmx.mbeanserver=org.graalvm.nativeimage.builder \
--add-exports=java.management/sun.management=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.api.replacements=jdk.graal.compiler.hostvmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.api.runtime=jdk.graal.compiler.hostvmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.core.target=jdk.graal.compiler.hostvmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.phases.util=jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.runtime=jdk.graal.compiler.hostvmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.util.json=com.oracle.graal.reporter \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto,org.graalvm.truffle.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=jdk.graal.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=jdk.graal.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.riscv64=jdk.graal.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=com.oracle.graal.graal_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta.annotation=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.configure,org.graalvm.nativeimage.guest.staging,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.configure,org.graalvm.nativeimage.guest.staging,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto,org.graalvm.truffle.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.riscv64=com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=com.oracle.graal.graal_enterprise,jdk.graal.compiler,jdk.graal.compiler.hostvmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.services=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.events=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.event=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.jfc=com.oracle.svm.svm_enterprise,org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.settings=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.tracing=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal=com.oracle.svm.svm_enterprise,org.graalvm.nativeimage.builder \
--add-exports=jdk.management/com.sun.management.internal=org.graalvm.nativeimage.builder \
-XX:+UseJVMCINativeLibrary \
-XX:MaxRAMPercentage=84.99999999642142 \
'-Dsvm.build.memoryUsageReasonText=13.27GiB of memory (85% of system memory because $CI set to '"'"'true'"'"')' \
-Xss10m \
-XX:+UseParallelGC \
-XX:GCTimeRatio=9 \
-XX:+ExitOnOutOfMemoryError \
-XX:+UseCompressedOops \
-Djava.awt.headless=true \
'-Dorg.graalvm.vendor=Oracle Corporation' \
-Dorg.graalvm.vendorurl=https://www.graalvm.org/ \
'-Dorg.graalvm.vendorversion=Oracle GraalVM 25.1.0-dev+10.1' \
-Dorg.graalvm.version=25.1.0-dev \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Djava.util.concurrent.ForkJoinPool.common.exceptionHandler=com.oracle.svm.hosted.CommonPoolUncaughtExceptionHandler \
'-Djava.util.concurrent.ForkJoinPool.common.threadFactory=com.oracle.svm.hosted.NativeImageSystemClassLoader$NativeImageForkJoinWorkerThreadFactory' \
-Dorg.graalvm.nativeimage.imagecode=buildtime \
-Xshare:off \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-Djava.lang.invoke.MethodHandle.DONT_INLINE_THRESHOLD=-1 \
-Djava.lang.invoke.MethodHandle.PROFILE_GWT=false \
-Djunit.platform.listeners.uid.tracking.output.dir=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/test-results/test/testlist \
-Dcom.oracle.svm.provided=/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-shared.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest-staging.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/vmaccess-guest.jar \
--add-modules=ALL-DEFAULT \
--enable-native-access=org.graalvm.nativeimage.guest.staging,org.graalvm.espresso.shared.svm,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.shared,org.graalvm.nativeimage.shaded.capnproto,org.graalvm.nativeimage.foreign,com.oracle.svm.svm_enterprise,com.oracle.graal.reporter,jdk.graal.compiler.vmaccess.guest,jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess,org.graalvm.nativeimage.configure,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--module-path \
/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/reporter.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/pointsto.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-shared.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/objectfile.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-configure.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/hostvmaccess.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-foreign.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/vmaccess.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/native-image-base.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-enterprise.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-capnproto-runtime.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-guest-staging.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/espresso-svm.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/vmaccess-guest.jar \
--module \
org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner \
-keepalive \
/proc/11270/comm \
-imagecp \
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/libs/io.mockk.mockk-agent-jvm_tests.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.mockk/mockk-agent-jvm/1.14.9/e1e1acc201dc6ec3bfb9d2d8f10e5960eda3c9f0/mockk-agent-jvm-1.14.9.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.mockk/mockk-core-jvm/1.14.9/79bd4a6018f7cf10b6b5cc539048804f1225cc21/mockk-core-jvm-1.14.9.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/2.2.21/17bc47f5dfe27c2351538e711f0afe377b0de6b2/kotlin-reflect-2.2.21.jar:/home/runner/.gradle/caches/modules-2/files-2.1/io.mockk/mockk-agent-api-jvm/1.14.9/a823c7aebc0396b4201c0d397ac352fad30c76d8/mockk-agent-api-jvm-1.14.9.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/2.3.20/ed866de74ad3d49086a27bbd75952cd186479436/kotlin-stdlib-2.3.20.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.graalvm.buildtools/junit-platform-native/1.0.0/2974f834a2f0a0105f2284fba7ef16dc4d60de56/junit-platform-native-1.0.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter/5.9.2/26c586fbe0ebd81b48c9f11f0d998124248697ae/junit-jupiter-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-reporting/1.9.2/8bea292e2a546e4c21071f6bee60917ffd9f0101/junit-platform-reporting-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-launcher/1.9.2/38e0bfad8c57d4cd1a8f27926c25ffe9543068d6/junit-platform-launcher-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.vintage/junit-vintage-engine/5.9.2/53421816bde124a564a64ba005dcc0c8e66a9722/junit-vintage-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-params/5.9.2/bc2765afb7b85b583c710dd259a11c6b8c39e912/junit-jupiter-params-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-engine/5.9.2/572f7a553b53f83ee59cc045ce1c3772864ab76c/junit-jupiter-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-api/5.9.2/fed843581520eac594bc36bb4b0f55e7b947dda9/junit-jupiter-api-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-engine/1.9.2/40aeef2be7b04f96bb91e8b054affc28b7c7c935/junit-platform-engine-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-commons/1.9.2/6f9f8621d8230cd38aa42e58ccbc0c00569131ce/junit-platform-commons-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-console/1.9.2/5548148dd55e820c42bd817ab93c1eaa05ce37f6/junit-platform-console-1.9.2.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/tck-test-support/build/libs/tck-test-support-1.0.0-SNAPSHOT.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.assertj/assertj-core/3.22.0/c300c0c6a24559f35fa0bd3a5472dc1edcd0111e/assertj-core-3.22.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.jetbrains/annotations/13.0/919f0dfe192fb4e063e7dacadee7f8bb9a2672a9/annotations-13.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.objenesis/objenesis/3.4/675cbe121a68019235d27f6c34b4f0ac30e07418/objenesis-3.4.jar:/home/runner/.gradle/caches/modules-2/files-2.1/net.bytebuddy/byte-buddy/1.18.2/7ac991b4bd502e2567efcdecc0d2e9b3f7dd3859/byte-buddy-1.18.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/net.bytebuddy/byte-buddy-agent/1.18.2/62f38a6faf7f069d661b79a07d566f504b0b20c4/byte-buddy-agent-1.18.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/junit/junit/4.13.2/8ac9e16d933b6fb43bc7f576336b8f4d7eb5ba12/junit-4.13.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.opentest4j/opentest4j/1.2.0/28c11eb91f9b6d8e200631d46e20a7f407f2a046/opentest4j-1.2.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/classes/java/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/classes/kotlin/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
-H:ReachabilityMetadataResources@file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/resources/test/META-INF/native-image/reachability-metadata.json+api=META-INF/native-image/reachability-metadata.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=io.mockk.mockk-agent-jvm_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jetbrains.kotlin/kotlin-stdlib/1.7.10,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jetbrains/annotations/13.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.mockk/mockk-agent-jvm/1.14.9,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.objenesis/objenesis/3.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/net.bytebuddy/byte-buddy-agent/1.10.9,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.mockk/mockk-agent-api-jvm/1.14.9,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jetbrains.kotlin/kotlin-reflect/1.7.10,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/io.mockk/mockk-core-jvm/1.14.9 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=0560e6d4-c470-25bd-66dd-894b409336e5 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'io.mockk.mockk-agent-jvm_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.6s @ 0.24GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [******]                                                                 (48.6s @ 1.32GiB)
   13,097 types,  15,812 fields, and  61,079 methods found reachable
      465 types,     100 fields, and   1,126 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      172 resource accesses registered with 331.01KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (7.0s @ 1.23GiB)
[4/8] Parsing methods...      [**]                                                                      (4.7s @ 1.32GiB)
[5/8] Inlining methods...     [***]                                                                     (2.6s @ 1.39GiB)
[6/8] Compiling methods...    [*****]                                                                  (27.2s @ 1.15GiB)
[7/8] Laying out methods...   [**]                                                                      (4.5s @ 1.45GiB)
[8/8] Creating image...       [**]                                                                      (4.5s @ 1.59GiB)
  14.10MiB (36.74%) for code area:    38,095 compilation units
  20.44MiB (53.25%) for image heap:  322,429 objects and 40 resources
   3.84MiB (10.01%) for other data
  38.38MiB in total image size, 35.13MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.55MiB o.g.n.runtime/com.oracle.svm.core.code              4.55MiB byte[] for code metadata
 738.04KiB java.base/java.util                                 4.17MiB byte[] for string data
 414.97KiB java.base/jdk.internal.classfile.impl               1.95MiB java.base/java.lang.String
 347.69KiB kotlin.reflect.jvm.internal.impl.metadata           1.71MiB java.base/java.lang.Class
 341.90KiB java.base/java.lang                                 1.40MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 271.42KiB java.base/com.sun.crypto.provider                   1.01MiB byte[] for general heap data
 247.59KiB net.bytebuddy.jar.asm                             575.80KiB byte[] for reflection metadata
 246.74KiB java.base/java.util.concurrent                    492.42KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 235.05KiB java.base/java.text                               378.04KiB java.base/java.lang.Object[]
 224.73KiB org.assertj.core.internal.bytebuddy.jar.asm       358.41KiB java.base/j.u.concurrent.ConcurrentHashMap$Node
   9.55MiB for 447 more packages                               3.88MiB for 2947 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 25 component(s) is embedded in binary (1.38KiB). 14 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       8.4s (7.6% of total time) in 1132 GCs | Peak RSS: 2.28GiB | CPU load: 3.65
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/native/nativeTestCompile/io.mockk.mockk-agent-jvm_tests-tests (executable, 35.13MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'io.mockk.mockk-agent-jvm_tests-tests' in 1m 49s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

io_mockk.mockk_agent_jvm.BootJarLoaderTest > initializesAgentFactoryWithDispatcherClassesOnBootstrapPath() FAILED

io_mockk.mockk_agent_jvm.ClassLoadingStrategyChooserTest > choosesClassLoadingStrategyForApplicationClass() SUCCESSFUL

io_mockk.mockk_agent_jvm.DefaultInterfaceMethodResolverInnerCompanionTest > resolvesKotlinInterfaceDefaultImplementationFromDefaultImplsClass() SUCCESSFUL

io_mockk.mockk_agent_jvm.ProxyMakerInnerCompanionTest > proxyCreationWithoutInlineInstrumentationScansInheritedFinalMethods() SUCCESSFUL

io_mockk.mockk_agent_jvm.ProxyMakerTest > createsProxyUsingTargetDefaultConstructor() FAILED


Failures (2):
  JUnit Jupiter:BootJarLoaderTest:initializesAgentFactoryWithDispatcherClassesOnBootstrapPath()
    MethodSource [className = 'io_mockk.mockk_agent_jvm.BootJarLoaderTest', methodName = 'initializesAgentFactoryWithDispatcherClassesOnBootstrapPath', methodParameterTypes = '']
    => java.lang.IllegalStateException: No compatible attachment provider is available
       net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:608)
       net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:592)
       net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:544)
       net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:521)
       io.mockk.proxy.jvm.JvmMockKAgentFactory.initInstrumentation(JvmMockKAgentFactory.kt:135)
       io.mockk.proxy.jvm.JvmMockKAgentFactory.init(JvmMockKAgentFactory.kt:39)
       io_mockk.mockk_agent_jvm.BootJarLoaderTest.initializesAgentFactoryWithDispatcherClassesOnBootstrapPath(BootJarLoaderTest.kt:22)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       [...]
  JUnit Jupiter:ProxyMakerTest:createsProxyUsingTargetDefaultConstructor()
    MethodSource [className = 'io_mockk.mockk_agent_jvm.ProxyMakerTest', methodName = 'createsProxyUsingTargetDefaultConstructor', methodParameterTypes = '']
    => java.lang.IllegalStateException: No compatible attachment provider is available
       net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:608)
       net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:592)
       net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:544)
       net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:521)
       io.mockk.proxy.jvm.JvmMockKAgentFactory.initInstrumentation(JvmMockKAgentFactory.kt:135)
       io.mockk.proxy.jvm.JvmMockKAgentFactory.init(JvmMockKAgentFactory.kt:39)
       io_mockk.mockk_agent_jvm.ProxyMakerTest.createsProxyUsingTargetDefaultConstructor(ProxyMakerTest.kt:26)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       [...]

Test run finished after 4 ms
[         7 containers found      ]
[         0 containers skipped    ]
[         7 containers started    ]
[         0 containers aborted    ]
[         7 containers successful ]
[         0 containers failed     ]
[         5 tests found           ]
[         0 tests skipped         ]
[         5 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         2 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.mockk/mockk-agent-jvm/1.14.9/build/native/nativeTestCompile/io.mockk.mockk-agent-jvm_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

BUILD FAILED in 1m 55s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.mockk:mockk-agent-jvm:1.14.9 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 57s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.14.9 at 2026-05-19T03:36:41Z with exit code 1 after 118s
❌ ERROR: 'native-image run' for 1.14.9 failed with exit code 1.
FAILED[native-image run][1.14.9][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.14.9" ./gradlew clean test -Pcoordinates="io.mockk:mockk-agent-jvm:1.14.9"]
```

--- a/tests/src/io.mockk/mockk-agent-jvm/1.14.9/src/test/kotlin/io_mockk/mockk_agent_jvm/MockkNativeImageSupport.kt
+++ b/tests/src/io.mockk/mockk-agent-jvm/1.14.9/src/test/kotlin/io_mockk/mockk_agent_jvm/MockkNativeImageSupport.kt
@@ -31,7 +31,10 @@ internal object MockkNativeImageSupport {
 
     private fun hasByteBuddyAttachmentFailure(throwable: Throwable): Boolean =
         throwable.causeSequence().any { current: Throwable ->
-            current is IllegalStateException && current.message?.startsWith("Error during attachment using:") == true
+            current is IllegalStateException && (
+                current.message?.startsWith("Error during attachment using:") == true ||
+                    current.message == "No compatible attachment provider is available"
+                )
         }
 
     private fun hasJavaDispatcherInitializationFailure(throwable: Throwable): Boolean =

--- a/tests/src/io.mockk/mockk-agent-jvm/1.14.9/src/test/kotlin/io_mockk/mockk_agent_jvm/MockkNativeImageSupportTest.kt
+++ b/tests/src/io.mockk/mockk-agent-jvm/1.14.9/src/test/kotlin/io_mockk/mockk_agent_jvm/MockkNativeImageSupportTest.kt
@@ -1,0 +1,37 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io_mockk.mockk_agent_jvm
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+public class MockkNativeImageSupportTest {
+    @Test
+    fun treatsByteBuddyAttachmentProviderFailureAsExpectedInNativeImageRuntime(): Unit {
+        withNativeImageRuntimeProperty {
+            val throwable: Throwable = IllegalStateException("No compatible attachment provider is available")
+
+            assertThat(MockkNativeImageSupport.isExpectedNativeImageFailure(throwable)).isTrue()
+        }
+    }
+
+    private fun withNativeImageRuntimeProperty(block: () -> Unit): Unit {
+        val key: String = "org.graalvm.nativeimage.imagecode"
+        val previousValue: String? = System.getProperty(key)
+        System.setProperty(key, "runtime")
+
+        try {
+            block()
+        } finally {
+            if (previousValue == null) {
+                System.clearProperty(key)
+            } else {
+                System.setProperty(key, previousValue)
+            }
+        }
+    }
+}

